services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AutoProvisionerConfigProperties.java

```diff
@@ -12,12 +12,13 @@
  *******************************************************************************/
 package org.eclipse.hono.deviceregistry.service.device;
 
+import org.eclipse.hono.config.ProtocolAdapterProperties;
 import org.eclipse.hono.util.RegistrationConstants;
 
 /**
  * Configuration properties for Hono's gateway-based auto-provisioning.
  */
-public class AutoProvisionerConfigProperties {
+public class AutoProvisionerConfigProperties extends ProtocolAdapterProperties {
```

Comment:
I see why this is necessary here but I have to admit that I do not like it :-(
IMHO we should address this in a subsequent PR ...

Reply:
Yes, I also stumbled upon this when looking at CommandRouter which actually isn't a real protocol adapter neither...